.github/actions/docker-build/action.yml

```diff
@@ -26,21 +31,23 @@ runs:
   using: composite
   steps:
     - name: Docker Build ${{ inputs.target }}
-      uses: addnab/docker-run-action@v3
-      with:
-        image: ${{ inputs.image }}
-        options: >
-          --privileged
-          --user 0:0
-          -v ${{ github.workspace }}/.cargo/git/db:/usr/local/cargo/git/db
-          -v ${{ github.workspace }}/.cargo/registry/cache:/usr/local/cargo/registry/cache
-          -v ${{ github.workspace }}/.cargo/registry/index:/usr/local/cargo/registry/index
-          -v ${{ github.workspace }}:/build
-          -w /build
-        run: |
+      shell: bash
+      run: |
+        code="
           set -e
           ${{ inputs.pre }}
           rustup target add ${{ inputs.target }}
           corepack enable
           RUST_TARGET=${{ inputs.target }} pnpm build:binding:${{ inputs.profile }}
           ${{ inputs.post }}
+        "
```

Comment:
Let's land this first in a PR prior this this PR, so the rest of this PR is revertible.